src/NumericalAlgorithms/LinearOperators/PowerMonitors.cpp

```diff
@@ -17,6 +17,39 @@
 
 namespace PowerMonitors {
 
+double detail::relative_truncation_error_impl(
+    const DataVector& input_power_monitors, const size_t upperBound) {
+  // upperBound needs to be smaller than the number of power monitors in the
+  // current dimension, i.e.
+  // upperBound < gsl::at(input_power_monitors, sliced_dim).size()
```

Comment:
`<=` I think? Also make this an assert: `ASSERT(num_modes <= ..., "message here")`

Also assert that num_modes and the size of the DataVector are both >= 2.